### PR TITLE
Fixes for PRE test

### LINKED ISSUE #1112: Figure out why PRE.*.ADESP tests run forever on desktops
Once the issue is resolved, re-enable this test in cime_developer suite.

### PR DESCRIPTION
**Pull request by gold2718**
New method for setting and detecting pause alarms in component models
 Refactor access to components which are paused
 Pause works with the driver and at least one active component (POP).
 Active component (POP) able to correctly process resume signal.
 PRE test refactored to be more flexible.
 Modify code and scripts to error if NINST_ESP > 1.
 Fixed NCK test to not set NINST_ESP > 1
 Added PRE test back to cime_developer.

Test suite: scripts_regression_tests.py on Yellowstone (CESM) and compute001 (ACME)
                   CESM prealpha on hobart_nag
Test baseline: NA
Test namelist changes: NA
Test status: bit for bit

Fixes #1112 

User interface changes?: NA

Code review:

### CODE REVIEW

File: scripts/lib/CIME/SystemTests/pre.py [hunk 1]
```diff
@@ -31,26 +33,36 @@ def __init__(self, case):
                                        run_two_suffix='pr',
                                        run_one_description='no pause/resume',
                                        run_two_description='pause/resume')
+        self._stopopt  = 'ndays'
```
Comment (jedwards4b) [inline scripts/lib/CIME/SystemTests/pre.py]: Should these be in config_tests.xml and not here?

File: src/drivers/mct/cime_config/buildnml [hunk 2]
```diff
@@ -23,6 +23,24 @@ from CIME.XML.files import Files
 logger = logging.getLogger(__name__)
 
 ###############################################################################
+def _get_time_in_seconds(time, unit):
```
Comment (jedwards4b) [inline src/drivers/mct/cime_config/buildnml]: Isn't there already a function in CIME.utils for this?  If not wouldn't that be a better place for it?  Perhaps all of the time utilities should be moved to a new time_utils.py file.

File: scripts/lib/CIME/SystemTests/pre.py (hunk 1 above)
Comment (gold2718) [inline scripts/lib/CIME/SystemTests/pre.py]: They were really placeholders. I modified the values to just initialize the variables in `__init__`, the working values do come from config_tests.xml.

File: src/drivers/mct/cime_config/buildnml (hunk 2 above)
Comment (gold2718) [inline src/drivers/mct/cime_config/buildnml]: I could not find such a function so I moved this function from buildnml to CIME.utils.

**Comment by gold2718** [comment]
I believe I have addressed @jedwards4b useful comments (and rerun scripts_regression_tests).

**Pull request merged**